sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -628,7 +656,7 @@ private void ensureSerializable(Set<Class<? extends PipelineOptions>> interfaces
     public PipelineOptions deserialize(JsonParser jp, DeserializationContext ctxt)
         throws IOException, JsonProcessingException {
       ObjectNode objectNode = (ObjectNode) jp.readValueAsTree();
-      ObjectNode optionsNode = (ObjectNode) objectNode.get("options");
+      ObjectNode optionsNode = (ObjectNode) checkNotNull(objectNode.get("options"));
```

Comment:
is this a behavior change? if so, document and add error message?

if not, but was added for testing, remove?